examples/seismic/source.py

```diff
@@ -71,17 +120,15 @@ class WaveletSource(PointSource):
     :param time: Discretized values of time in ms
     """
 
-    def __new__(cls, *args, **kwargs):
-        time = kwargs.get('time')
+    def __new__(self, *args, **kwargs):
         npoint = kwargs.get('npoint', 1)
-        kwargs['ntime'] = len(time)
         kwargs['npoint'] = npoint
-        obj = PointSource.__new__(cls, *args, **kwargs)
+        obj = PointSource.__new__(self, *args, **kwargs)
 
-        obj.time = time
         obj.f0 = kwargs.get('f0')
+        _time = obj.time
```

Comment:
unnecessary?

Reply:
Yes - this invokes linspace and the following loop where its used could be large.

Reply:
perhaps a symptom that you should/could use a `cached_property` for time

Reply:
Good point.